src/lib/cspauth/csp_auth.go

```diff
@@ -0,0 +1,95 @@
+package cspauth
+
+import (
+	"context"
+	"fmt"
+	"github.com/vmware-tanzu/cloud-native-security-inspector/src/lib/log"
+	"github.com/vmware-tanzu/cloud-native-security-inspector/src/lib/retry"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"math"
+	"sigs.k8s.io/controller-runtime"
+	"time"
+)
+
+const (
+	tokenMaxAgeSeconds   = 600
+	providerAccessSecret = "accessSecret"
+)
+
+// Provider is an interface to interact with an authorization service
+type Provider interface {
+	// GetBearerToken retrieves a short-lived access token to use in a single HTTP request
+	GetBearerToken(context.Context) (string, error)
+}
+
+type cspAuth struct {
+	apiToken  string
+	cspClient CSPClient
+
+	currentToken string
+	expiration   time.Time
+}
+
+func (a *cspAuth) GetBearerToken(ctx context.Context) (string, error) {
+	if a.currentToken == "" || time.Now().After(a.expiration) {
+		if err := a.refreshToken(ctx); err != nil {
+			return "", err
+		}
+	}
+	return a.currentToken, nil
+}
+
+func (a *cspAuth) refreshToken(ctx context.Context) error {
+	return retry.NewRetry(
+		retry.WithName("auth token refresh"),
+		retry.WithMaxAttempts(3),
+		retry.WithIncrementDelay(5*time.Second, 5*time.Second),
+	).Run(ctx, func() (bool, error) {
+		now := time.Now()
+		cspAuthResponse, err := a.cspClient.GetCspAuthorization(ctx, a.apiToken)
+		if err != nil {
+			fmt.Printf("We got an error back from CSP %s", err)
```

Comment:
Shouldn't we use a `log.error` like in other classes? This applies for every `fmt.Printf` in this file

Reply:
Done.